tests/integration/utils.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+# SPDX-License-Identifier: MIT
+
+import requests
+from requests_kerberos import HTTPKerberosAuth
+
+
+class Response:
+    """Contains simplified HTTP response
+
+    :attribute dict data: Data fetched from the Cachito API
+    :attribute int id: ID of the request in Cachito
+    :attribute int status: HTTP status code
+    """
+    def __init__(self, data, id, status):
+        self.data = data
+        self.id = id
+        self.status = status
+
+
+class Client:
+    """Class that simplifies access to the Cachito API
+
+    :attribute string _cachito_api_url: URL of the Cachito API
+    """
+    def __init__(self, cachito_api_url):
+        self._cachito_api_url = cachito_api_url
+
+    def fetch_all_requests(self):
+        """Fetch a list of requests from the Cachito API
+
+        :return: Object that contains response from Cachito API
+        :rtype: Response
+        """
+        resp = requests.get(f"{self._cachito_api_url}requests")
+        resp.raise_for_status()
+        return Response(resp.json(), None, resp.status_code)
+
+    def fetch_request(self, request_id):
+        """Fetch a request from Cachito API
+
+        :param request_id: ID of the request in Cachito
+        :return: Object that contains response from Cachito API
+        :rtype: Response
+        """
+        resp = requests.get(f"{self._cachito_api_url}requests/{request_id}")
+        return Response(resp.json(), resp.json()["id"], resp.status_code)
+
+    def create_new_request(self, **kwargs):
+        """Create new request in Cachito
+
+        :param dict kwargs: Payload for request
+        :return: Object that contains response from Cachito API
+        :rtype: Response
+        """
+        resp = requests.post(f"{self._cachito_api_url}requests", auth=HTTPKerberosAuth(),
```

Comment:
I'd like if `auth` was configurable to either use Kerberos, SSL Certs, or no auth. This would make the tests more portable between the various environment. The local dev environment, `docker-compose up`, does not use auth, for example.

Reply:
done

Reply:
I think this would need a bit more changes to make it truly configurable. For instance, it would be ideal if alongside `api_url` in the config yaml, we had `api_auth_type` which could be set to either 'kerberos', or `null` (we can add `ssl` later). Then, in `create_new_request`, we'd map the string `'kerberos'` to 'HTTPKerberosAuth()', for example.

This can be done later in a follow up PR.

Reply:
I did the changes, can you check that? I am not sure if there's a better way to do the mapping.

Reply:
In python, `None` is a hashable value. So you could have:
```python
authentication_mapping = {
  None: None,
  "kerberos": HTTPKerberosAuth(),
}
```
This is better than having to force a string of value `'null``.

But there's no point to map a value to itself. Consider this:
```python
authentication_mapping = {
  "kerberos": HTTPKerberosAuth()
}
```

Then use `get` with a default value:
```python
auth=authentication_mapping.get(auth, auth),
```

Reply:
Thanks for the explanation, it's changed now.